rules/REQUEST-933-APPLICATION-ATTACK-PHP.conf

```diff
@@ -436,12 +436,12 @@ SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_H
 #   \(.*\)
 #       Parentheses optionally containing function parameters
 #
-SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_FILENAME|ARGS_NAMES|ARGS|XML:/* "@rx \$+(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*|\s*{.+})(?:\s|\[.+\]|{.+}|/\*.*\*/|//.*|#.*)*\(.*\)" \
+SecRule REQUEST_COOKIES|!REQUEST_COOKIES:/__utm/|REQUEST_COOKIES_NAMES|REQUEST_FILENAME|ARGS_NAMES|ARGS|XML:/* "@rx \$+(?:[a-zA-Z_\x7f-\xff][a-zA-Z0-9_\x7f-\xff]*|{.+}).*\(.*\)" \
     "id:933180,\
     phase:2,\
     block,\
     capture,\
-    t:none,\
+    t:none,t:compressWhitespace,t:removeWhitespace,t:removeCommentsChar,\
```

Comment:
`removeWhitespace` makes `compressWhitespace` unnecessary, right?

Reply:
it's not clear on modsec documentation if removeWhitespace removes also \n. we do know that compressWhitespace replace \n with a space \x20

Reply:
Yeah, I wondered about that too :/

Reply:
I'm not sure if this will be helpful anymore, but maybe it will bee :smile: 

#### ModSec 2:

```
if (isspace(input[i])||(input[i] == NBSP))
```

#### ModSec 3:

```
if (std::isspace(static_cast<unsigned char>(value[i]))                   
            || (value[i] == nonBreakingSpaces)                                   
            || value[i] == nonBreakingSpaces2)
```

The `isspace` function: “Checks if the given character is a whitespace character, i.e. either space (0x20), form feed (0x0c), line feed (0x0a), carriage return (0x0d), horizontal tab (0x09) or vertical tab (0x0b).”

So `removeWhitespace` appears to function on the characters listed in the previous sentence, plus one or two non-breaking spaces :)